src/java/org/apache/cassandra/metrics/TableMetrics.java

```diff
@@ -1173,32 +1204,55 @@ private LatencyMetrics createLatencyMetrics(String namePrefix, LatencyMetrics ..
      * @return true if first time metric with that name has been registered
      */
     private boolean register(String name, String alias, Metric metric)
+    {
+        return register(name, alias, null, metric);
+    }
+
+    /**
+     * Registers a metric to be removed when unloading CF.
+     * @param deprecated the deprecated name for the metric which should be {@code null} if no deprecation is required
+     * @return true if first time metric with that name has been registered
+     */
+    private boolean register(String name, String alias, String deprecated, Metric metric)
     {
         boolean ret = allTableMetrics.putIfAbsent(name, ConcurrentHashMap.newKeySet()) == null;
         allTableMetrics.get(name).add(metric);
-        all.add(() -> releaseMetric(name, alias));
+        all.add(() -> releaseMetric(name, alias, deprecated));
         return ret;
     }
 
-    private void releaseMetric(String metricName, String metricAlias)
+    private void releaseMetric(String metricName, String metricAlias, String metricDeprecated)
     {
         CassandraMetricsRegistry.MetricName name = factory.createMetricName(metricName);
-        CassandraMetricsRegistry.MetricName alias = aliasFactory.createMetricName(metricAlias);
+
         final Metric metric = Metrics.getMetrics().get(name.getMetricName());
         if (metric != null)
-        {   // Metric will be null if we are releasing a view metric.  Views have null for ViewLockAcquireTime and ViewLockReadTime
+        {
+            // Metric will be null if we are releasing a view metric.  Views have null for ViewLockAcquireTime and ViewLockReadTime
             allTableMetrics.get(metricName).remove(metric);
-            Metrics.remove(name, alias);
+
+            CassandraMetricsRegistry.MetricName[] aliases = null == metricDeprecated ? new CassandraMetricsRegistry.MetricName[1] : new CassandraMetricsRegistry.MetricName[3];
```

Comment:
nit: I'd probably just make two calls to `Metrics.remove()` here rather than worry about array creation with a conditional size.